manifests/etcdquorumguard/deployment.yaml

```diff
@@ -0,0 +1,88 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: etcd-quorum-guard
+  namespace: kube-system
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      name: etcd-quorum-guard
+  strategy:
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: etcd-quorum-guard
+        app: etcd-quorum-guard
+    spec:
+      hostNetwork: true
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - "etcd-quorum-guard"
+            topologyKey: kubernetes.io/hostname
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      priorityClassName: "system-cluster-critical"
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+        operator: Exists
+      - key: node.kubernetes.io/memory-pressure
+        effect: NoSchedule
+        operator: Exists
+      - key: node.kubernetes.io/disk-pressure
+        effect: NoSchedule
+        operator: Exists
+      - key: node.kubernetes.io/not-ready
+        effect: NoExecute
+        operator: Exists
+      - key: node.kubernetes.io/unreachable
+        effect: NoExecute
+        operator: Exists
+      - key: node.kubernetes.io/unschedulable
+        effect: NoExecute
+        operator: Exists
+      containers:
+      - image: registry.svc.ci.openshift.org/openshift/origin-v4.0:base
+        imagePullPolicy: IfNotPresent
+        name: etcd-quorum-guard-container
+        volumeMounts:
+        - mountPath: /mnt/kube
+          name: kubecerts
+        command:
+        - "/bin/sleep"
+        args:
+        - "infinity"
+        readinessProbe:
+          exec:
+            command:
+            - /bin/sh
+            - -c
+            - |
+                declare -r croot=/mnt/kube
+                declare -r health_endpoint="https://127.0.0.1:2379/health"
+                declare -r cert="$(find $croot -name 'system:etcd-peer*.crt' -print -quit)"
+                declare -r key="${cert%.crt}.key"
+                declare -r cacert="$croot/ca.crt"
+                [[ -z $cert || -z $key ]] && exit 1
+                curl --max-time 2 --silent --cert "${cert//:/\:}" --key "$key" --cacert "$cacert" "$health_endpoint" |grep '{ *"health" *: *"true" *}'
```

Comment:
/cc @hexfusion 
please use the metrics client certs that were created to connect to etcd

Reply:
Where are those certs located?

Reply:
You could get crt/key with something like

```
for ext in crt key; do oc -n openshift-config get secrets etcd-metric-client -o yaml | grep  tls.${ext} | awk '{ print $2 }' | base64 --decode > etcd-metrics-client.${ext}; done;
```

ca

oc get configmap -n openshift-config admin-kubeconfig-client-ca -o yaml

Reply:
you can use etcd proxy for /health with these certs. port `9979` `2379`

Reply:
I can do that inside the pod?

Reply:
> I can do that inside the pod?

Not following the certs are not file on disk can you consume them via configmap/secret?

Reply:
The certs do appear to be on disk on the master, in `/etc/kubernetes/static-pod-resources/etcd-member`.  If I try to run `oc -n openshift-config get secrets etcd-metric-client -o yaml` on the node, it fails (missing or incomplete configuration info).  Not to mention that `jq` is not present in the standard image.

Reply:
I was just giving reference to the namespace and name of the resources.

We want to use the metrics certs because they have a different chain of trust and do not allow access to the KV store directly. The certs you see on disk are the metric proxy server certs the key usage does show `TLS Web Server Authentication` as well as `TLS Web Client Authentication`. The latter is probably a bug we should remove. But it does not have 127.0.0.1 in the IP SAN so you could not use it the way you do with the etcd client cert.

From a security point of view, we would like to use metrics client certs MCO might have to sync those into this namespace so they can be mounted and consumed here.

Reply:
I agree with the rationale; my question is how to get the appropriate cert.

Reply:
working on this now

Reply:
https://github.com/openshift/machine-config-operator/pull/623

Reply:
Note that the etcd-quorum-guard proper does not have any Go code in it; it's simply (right now) a static deployment and disruption budget, with the lone pod being a trivial script.

Reply:
with #623 you should be able to mount the resources and then consume in your bash as local files. Something like.

```yaml
        volumeMounts:
          - mountPath: "/etc/ssl/certs/etcd"
            name: etcd-metric-client
            readOnly: true
      volumes:
        - name: etcd-metric-client
          secret:
            secretName: etcd-metric-client
```